backend/oasst_backend/user_repository.py

```diff
@@ -1,16 +1,40 @@
 from typing import Optional
+from uuid import UUID
 
 from oasst_backend.models import ApiClient, Message, User
+from oasst_shared.exceptions import OasstError, OasstErrorCode
 from oasst_shared.schemas import protocol as protocol_schema
 from oasst_shared.schemas.protocol import LeaderboardStats
 from sqlmodel import Session, func
+from starlette.status import HTTP_403_FORBIDDEN, HTTP_404_NOT_FOUND
 
 
 class UserRepository:
     def __init__(self, db: Session, api_client: ApiClient):
         self.db = db
         self.api_client = api_client
 
+    def query_frontend_user(
+        self, auth_method: str, username: str, api_client_id: Optional[UUID] = None
+    ) -> Optional[User]:
+        if not api_client_id:
+            api_client_id = self.api_client.id
+
+        if not self.api_client.trusted and api_client_id != self.api_client.id:
```

Comment:
In which case can we have an api client id that is not the same as the self.api_client_id?

Reply:
Trusted api_clients can see/edit all users. Normally they would probably not go through the frontend-user query but theoretically this is possible.

Reply:
Okay thanks a lot!